projects/eslint-plugin-graphql/eslint/parser.js

```diff
@@ -1,65 +1,114 @@
 'use strict';
 
-const { parse, visit } = require('graphql');
-const gqlToEs = require('./gql-node-to-es-node');
+const { parse, buildSchema } = require('graphql');
+const Traverser = require('eslint/lib/shared/traverser');
+const path = require('path');
+const fs = require('fs');
 
-module.exports.parseForESLint = function(code, options) {
-  let ast;
+const toEslintAST = require('./to-eslint-ast');
+// TODO: verify this is the required set. Note: if something isn't here, it
+// will "auto visit". But specifying it here, will prevent traversal of unexpected members
+const visitorKeys = {
+  "Document": [ "definitions" ],
+  "FragmentDefinition": ["typeCondition","directives","selectionSet"],
+  "NamedType": [],
+  "SelectionSet": ["selections"],
+  "OperationDefinition": ["variableDefinitions","directives","selectionSet"],
+  "Field": ["alias","arguments","directives","selectionSet"],
+  "FragmentSpread": ["directives"],
+  "FieldDefinition": ["name", "arguments", "directives"]
+};
 
+module.exports.parseForESLint = function(code, options = {}) {
+  let graphqlAST;
   const filePath = typeof options === 'object' && options !== null && options.filePath || '<unknown file>';
   try {
-    ast = parse(code, options)
+    graphqlAST = parse(code, options)
   } catch(e) {
     e.message = `${e.message}\n Could not parse file: ${filePath}\n File Contents:\n ${code}\n`;
     e.file = filePath;
     throw e;
   }
 
-  // turn graphql AST into eslint compatible AST
-  visit(ast, { enter: gqlToEs });
+  let schemaString;
+  let schema;
+  let typeInfo;
+
+  if (options.schema) {
+    // if we are given an absolute path, automatically just read it
+    if (path.isAbsolute(options.schema)) {
+      schemaString = fs.readFileSync(options.schema, 'UTF8')
+    } else {
+      // otherwise resolve, that enables us to resolve relative the path of the file
+      schemaString = fs.readFileSync(resolve(options.schema, {
+        basedir: path.dirname(options.filePath)
+      }), 'UTF8');
```

Comment:
require.resolve

Reply:
what do we resolve relative to?

Reply:
throwing a nice error that tells the user to put `${__dirname__}/foo/bar` would be fine too.  The main thing is I expect users to configure `eslintrc` with relative paths and if they need to do something else, the dx for this case should be a "pit of success" style.